src/textual/widgets/_collapsible.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from textual.widget import Widget
+
+from .. import events, on
+from ..app import ComposeResult
+from ..containers import Container, Horizontal
+from ..message import Message
+from ..reactive import reactive
+from ..widget import Widget
+from ._label import Label
+
+__all__ = ["Collapsible"]
+
+
+class Collapsible(Widget):
+    """A collapsible container."""
+
+    collapsed = reactive(True)
+
+    DEFAULT_CSS = """
+    Collapsible {
+        width: 100%;
+        height: auto;
+    }
+    """
+
+    class Title(Horizontal):
+        DEFAULT_CSS = """
+        Title {
+            width: 100%;
+            height: auto;
+        }
+
+        Title:hover {
+            background: grey;
+        }
+
+        Title .label {
+            padding: 0 0 0 1;
+        }
+
+        Title #collapsed-symbol {
+            display:none;
+        }
+
+        Title.-collapsed #expanded-symbol {
+            display:none;
+        }
+
+        Title.-collapsed #collapsed-symbol {
+            display:block;
+        }
+        """
+
+        def __init__(
+            self,
+            *,
+            label: str,
+            collapsed_symbol: str,
+            expanded_symbol: str,
+            name: str | None = None,
+            id: str | None = None,
+            classes: str | None = None,
+            disabled: bool = False,
+        ) -> None:
+            super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+            self.collapsed_symbol = collapsed_symbol
+            self.expanded_symbol = expanded_symbol
+            self.label = label
+
+        class Toggle(Message):
+            """Request toggle."""
+
+        async def _on_click(self, event: events.Click) -> None:
+            """Inform ancestor we want to toggle."""
+            self.post_message(self.Toggle())
```

Comment:
I understood what you meant! I added `event.stop()` above. Is that sufficient?

Reply:
It still seems like bubbling `Toggle` or click messages when I tested `tests/snapshot_tests/snapshot_apps/collapsible_nested.py`...
How do I prevent this...? 🥲

Reply:
Ahhh, that's a fun one :D 

Well, you did the correct thing: `event.stop` prevents the `Click` from bubbling.
Now, the handler `_update_collapsed`, which could also just be `_on_title_toggle`, will receive a `Toggle` event, right?
And it uses that event to know the contents need to be toggled...
And after doing that, you should `.stop` that event too!
Otherwise it affects nested collapsibles, as you found out :) 
Does that make sense to you?

Reply:
Right! I think I got what you meant. Thank you! I just pushed the fixes.